### PR TITLE
Add TextAnalyticsInput per arch board feedback

### LINKED ISSUE #9360: Add CONTRIBUTING.md for Text Analytics
See https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/appconfiguration/CONTRIBUTING.md as an example

### LINKED ISSUE #10285: Consider naming around DetectLanguageInput
Concern: What if we have more APIs that want to take text with ID? i.e. should this type be called `TextAnalyticsInput`?

It would be good if this type inherited from `TextAnalyticsInput`

https://apiview.dev/Assemblies/Review/b196a5e53fa2441f92e3b1accefa35e7

### PR DESCRIPTION
**Pull request by maririos**
- Add `TextAnalyticsInput` per arch board feedback. Fixes: https://github.com/Azure/azure-sdk-for-net/issues/10285
- Add general contributing guide to readme. Fixes https://github.com/Azure/azure-sdk-for-net/issues/9360

### CODE REVIEW

File: sdk/textanalytics/Azure.AI.TextAnalytics/CHANGELOG.md [hunk 1]
```diff
@@ -3,7 +3,7 @@
 ## 1.0.0-preview.4 (Unreleased)
 ### Breaking changes
 - Replaced `TextAnalyticsApiKeyCredential` with `AzureKeyCredential`.
-- Rename all input parameters `inputText` to `document`, and `inputs` to `documents`.
+- Renamed all input parameters `inputText` to `document`, and `inputs` to `documents`.
```
Comment (AlexanderSher) [inline sdk/textanalytics/Azure.AI.TextAnalytics/CHANGELOG.md]: Shouldn't we reflect the change in the type hierarchy here?

Comment (maririos) [reply sdk/textanalytics/Azure.AI.TextAnalytics/CHANGELOG.md]: Good point. I was thinking about that and wasn't sure.
So should I include it even if it is not a breaking change?

Comment (AlexanderSher) [reply sdk/textanalytics/Azure.AI.TextAnalytics/CHANGELOG.md]: Yes, we aren't limited only to breaking changes.

File: sdk/textanalytics/Azure.AI.TextAnalytics/src/DetectLanguageInput.cs [hunk 2]
```diff
@@ -44,10 +35,5 @@ public DetectLanguageInput(string id, string text)
         /// To remove this behavior, set to <see cref="None"/>.
         /// </summary>
         public string CountryHint { get; set; }
```
Comment (AlexanderSher) [inline sdk/textanalytics/Azure.AI.TextAnalytics/src/DetectLanguageInput.cs]: Not part of this change, but why `CountryHint` is a get/set property and `Id` is a read-only property?

File: sdk/textanalytics/Azure.AI.TextAnalytics/CHANGELOG.md (hunk 1 above)
Comment (maririos) [reply sdk/textanalytics/Azure.AI.TextAnalytics/CHANGELOG.md]: Thank you! I will update

File: sdk/textanalytics/Azure.AI.TextAnalytics/src/DetectLanguageInput.cs (hunk 2 above)
Comment (maririos) [reply sdk/textanalytics/Azure.AI.TextAnalytics/src/DetectLanguageInput.cs]: We want people to only be able to set `Id` and `Text` once when creating the `DetectLanguageInput` object. These are the properties that make an object unique.
`CountryHint` is not mandatory and could be added or modified whenever

**Review by pakrym — APPROVED** [review]
